### PR TITLE
Make two buttons possible/avoid AJAX refreshes CSRF

### LINKED ISSUE #92: AJAX request refreshes token, although no button on the page
When an AJAX request is loaded, the Facebook/google csrf token is regenerated, because it is done on `_construct` and therefore even called, if the block is not rendered

### LINKED ISSUE #93: Two buttons have different CSRF tokens
We want to add two FB buttons on the page, but the second button regenerates the CSRF token, so the first button doesn'T work anymore

### PR DESCRIPTION
**Pull request by Schrank**
FIXES #93 
FIXES #92 

Make it possible to have more than one button per type on the page and
make sure that an AJAX request does not refresh the CSRF token if the
button is not shown.

@Marko-M If I have some time over, I'm happy to dig through the issues and clean them up a bit. Not sure how much time I have to offer, but maybe better than currently 😅

### CODE REVIEW

**Comment by Schrank** [comment]
@sprankhub Review? 😅

File: app/code/community/Inchoo/SocialConnect/Block/Facebook/Button.php [hunk 1]
```diff
@@ -35,16 +40,24 @@ protected function _construct() {
...
         Mage::getSingleton('customer/session')
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/facebook/button.phtml');
     }
 
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getFacebookCsrf();
```
Comment (sprankhub) [inline app/code/community/Inchoo/SocialConnect/Block/Facebook/Button.php]: Err... `facebook_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?

File: app/code/community/Inchoo/SocialConnect/Block/Google/Button.php [hunk 2]
```diff
@@ -35,16 +40,26 @@ protected function _construct() {
...
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/google/button.phtml');
     }
 
+
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getGoogleCsrf();
```
Comment (sprankhub) [inline app/code/community/Inchoo/SocialConnect/Block/Google/Button.php]: Err... `google_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?

File: app/code/community/Inchoo/SocialConnect/Block/Linkedin/Button.php [hunk 3]
```diff
@@ -35,16 +40,24 @@ protected function _construct() {
...
         Mage::getSingleton('customer/session')
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/linkedin/button.phtml');
     }
 
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getLinkedinCsrf();
```
Comment (sprankhub) [inline app/code/community/Inchoo/SocialConnect/Block/Linkedin/Button.php]: Err... `linkedin_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?

**Review by sprankhub — CHANGES_REQUESTED** [review]

File: app/code/community/Inchoo/SocialConnect/Block/Facebook/Button.php (hunk 1 above)
Comment (Schrank) [reply app/code/community/Inchoo/SocialConnect/Block/Facebook/Button.php]: 🤦‍♂

**Comment by Schrank** [comment]
@sprankhub too much copied code around with modman. Thanks for heads up

**Review by sprankhub — APPROVED** [review]

**Pull request merged**